pkg/virt-controller/watch/dra/dra.go

```diff
@@ -0,0 +1,717 @@
+		return
+	}
+
+	curControllerRef := controller.GetControllerOf(curPod)
+	oldControllerRef := controller.GetControllerOf(oldPod)
+	controllerRefChanged := !equality.Semantic.DeepEqual(curControllerRef, oldControllerRef)
+	if controllerRefChanged {
+		// The ControllerRef was changed. Sync the old controller, if any.
+		if vmi := c.resolveControllerRef(oldPod.Namespace, oldControllerRef); vmi != nil {
+			c.enqueueVirtualMachine(vmi)
+		}
+	}
+
+	vmi := c.resolveControllerRef(curPod.Namespace, curControllerRef)
+	if vmi == nil {
+		return
+	}
+	c.enqueueVirtualMachine(vmi)
+}
+
+// resolveControllerRef returns the controller referenced by a ControllerRef,
+// or nil if the ControllerRef could not be resolved to a matching controller
+// of the correct Kind.
+func (c *DRAStatusController) resolveControllerRef(namespace string, controllerRef *metav1.OwnerReference) *v1.VirtualMachineInstance {
+	if controllerRef != nil && controllerRef.Kind == "Pod" {
+		// This could be an attachment pod, look up the pod, and check if it is owned by a VMI.
+		obj, exists, err := c.podIndexer.GetByKey(namespace + "/" + controllerRef.Name)
+		if err != nil {
+			return nil
+		}
+		if !exists {
+			return nil
+		}
+		pod, _ := obj.(*k8sv1.Pod)
+		controllerRef = controller.GetControllerOf(pod)
+	}
+	// We can't look up by UID, so look up by Name and then verify UID.
+	// Don't even try to look up by Name if it is nil or the wrong Kind.
+	if controllerRef == nil || controllerRef.Kind != v1.VirtualMachineInstanceGroupVersionKind.Kind {
+		return nil
+	}
+	vmi, exists, err := c.vmiIndexer.GetByKey(namespace + "/" + controllerRef.Name)
+	if err != nil {
+		return nil
+	}
+	if !exists {
+		return nil
+	}
+
+	if vmi.(*v1.VirtualMachineInstance).UID != controllerRef.UID {
+		// The controller we found with this Name is not the same one that the
+		// ControllerRef points to.
+		return nil
+	}
+	return vmi.(*v1.VirtualMachineInstance)
+}
+
+func (c *DRAStatusController) Run(threadiness int, stopCh <-chan struct{}) {
+	defer controller.HandlePanic()
+	defer c.Queue.ShutDown()
+	log.Log.Info("Starting DRA Status controller")
+
+	// Wait for cache sync before we start the pod controller
+	cache.WaitForCacheSync(stopCh, c.hasSynced)
+
+	// Start the actual work
+	for i := 0; i < threadiness; i++ {
+		go wait.Until(c.runWorker, time.Second, stopCh)
+	}
+
+	<-stopCh
+	log.Log.Info("Stopping DRA Status controller")
+}
+
+func (c *DRAStatusController) runWorker() {
+	for c.Execute() {
+	}
+}
+
+var draStatusControllerWorkQueueTracer = &traceUtils.Tracer{Threshold: time.Second}
+
+func (c *DRAStatusController) Execute() bool {
+	if !c.clusterConfig.GPUsWithDRAGateEnabled() && !c.clusterConfig.HostDevicesWithDRAEnabled() {
+		return false
+	}
+	key, quit := c.Queue.Get()
+	if quit {
+		return false
+	}
+
+	draStatusControllerWorkQueueTracer.StartTrace(key, "dra-status-controller VMI workqueue", trace.Field{Key: "Workqueue Key", Value: key})
```

Comment:
Isn't this covered by queue metrics already?

Reply:
WorkDurationMetric

Reply:
The metrics and tracing solve different purposes, it was explicitly requested to be added in review: https://github.com/kubevirt/kubevirt/pull/14365#discussion_r2150463495

Reply:
What I mean is that we seem to only trace the `updateStatus` and this would be covered by the metric imho